HerPortal.ManagementShell/AdminAction.cs

```diff
@@ -0,0 +1,92 @@
+using HerPortal.BusinessLogic.Models;
+
+namespace HerPortal.ManagementShell;
+
+public class AdminAction
+{
+    private readonly IDatabaseOperation dbOperation;
+    private readonly IOutputProvider outputProvider;
+    private readonly Dictionary<string, string> custodianCodeToLaDict = LocalAuthorityData.LocalAuthorityNamesByCustodianCode;
+
+    public AdminAction(IDatabaseOperation dbOperation, IOutputProvider outputProvider)
+    {
+        this.dbOperation = dbOperation;
+        this.outputProvider = outputProvider;
+    }
+
+    public User? GetUser(string emailAddress)
+    {
+        var portalUsers = dbOperation.GetUsersWithLocalAuthorities();
+        return
+            portalUsers.SingleOrDefault(user => string.Equals
+            (
+                user.EmailAddress,
+                emailAddress,
+                StringComparison.CurrentCultureIgnoreCase
+            ));
+    }
+
+    public bool ConfirmCustodianCodes(string[] codes, string userEmailAddress)
+    {
+        outputProvider.Output("ATTENTION! POTENTIAL DATA BREACH! Make sure below user should have the permissions you are about to grant");
+        outputProvider.Output(
+            $"You are changing permissions for user {userEmailAddress} for the following local authorities: ");
+
+        foreach (var code in codes)
+
+        {
+            var localAuthority = custodianCodeToLaDict[code];
+            outputProvider.Output($"{code}: {localAuthority}");
```

Comment:
Can we handle the case where a code doesn't match any LA?

Reply:
Amended - will now catch any exception, output the message and return false